assets/src/settings-page/analytics.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useContext, useEffect, useRef } from '@wordpress/element';
+import { Icon, plus, trash } from '@wordpress/icons';
+import { Button, TextControl, PanelRow, BaseControl } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { AMPDrawer } from '../components/amp-drawer';
+import { Options } from '../components/options-context-provider';
+
+/**
+ * Component for a single analytics entry.
+ *
+ * @param {Object} props Component props.
+ * @param {string} props.entryId Unique ID for the entry.
+ * @param {Function} props.onChange Callback to run when data changes.
+ * @param {Function} props.onDelete Callback to run when the entry is to be deleted.
+ * @param {string} props.type The entry type.
+ * @param {string} props.config The config JSON string.
+ */
+function AnalyticsEntry( { entryId, onChange, onDelete, type = '', config = '{}' } ) {
+	/**
+	 * Track the validity of the config JSON object. A nonempty custom validity string will block form submission.
+	 */
+	const textAreaRef = useRef();
+	useEffect( () => {
+		if ( ! textAreaRef?.current ) {
+			return;
+		}
+
+		if ( ! config ) {
+			textAreaRef.current.setCustomValidity( '' );
+			return;
+		}
+		try {
+			const parsedValue = JSON.parse( config );
+			if ( null === parsedValue || typeof parsedValue !== 'object' || Array.isArray( parsedValue ) ) {
+				textAreaRef.current.setCustomValidity( __( 'A JSON object is required, e.g. {…}', 'amp' ) );
+			} else {
+				textAreaRef.current.setCustomValidity( '' );
+			}
+		} catch ( e ) {
+			textAreaRef.current.setCustomValidity( e.message );
+		}
+	}, [ config ] );
+
+	/**
+	 * On creation, focus the cursor in the first input in the entry.
+	 */
+	const inputWrapper = useRef();
+	useEffect( () => {
+		if ( ! inputWrapper?.current ) {
+			return;
+		}
+
+		const firstInput = inputWrapper.current.querySelector( 'input' );
+		if ( firstInput ) {
+			firstInput.focus();
```

Comment:
This does not seem to work as intended when the drawer is first opened; the last input is focused instead of the first.

Reply:
Yes, I just saw the same. While I appreciate the intention, I don't think it's necessary in this context. Removed in 877f1f8d3457b30d73365587f368ee3b27bd7e52.